COMPACT_DENYLIST_FORMAT.md

```diff
@@ -0,0 +1,484 @@
+multihashes. This rule does not block subpaths that start at this CID, only
+the CID itself.
+
+Blocking layer recommendation: BlockService.
+
+##### `/ipfs/CID/PATH`
+
+IPFS-Path-Rule: Blocks the exact ipfs path that is referenced from the
+multihash embedded in the CID before attempting to resolve it. It does not block
+the CID that the path resolves to.
+
+Note `/ipfs/CID/path` and `/ipfs/CID/path/` are equivalent rules.
+
+Blocking layer recommendation: PathResolver.
+
+##### `/ipfs/CID/*` `/ipfs/CID/P/A/T/H*`
+
+IPFS-Path-Prefix-Rule: Blocks any multihash-path combination starting with the
+the given path prefix. `/*` includes the empty path. Thus `/ipfs/CID/*`
+blocks the CID itself, and any paths. Examples:
+
+- `/ipfs/CID/*` : blocks CID (by multihash) and any path before resolving.
+- `/ipfs/CID/ab*`: blocks any path derived from the CID (multihash) and starting with "ab", including "ab"
+- `/ipfs/CID/ab/*`: equivalent to the above.
+
+Blocking layer recommendation: PathResolver + (BlockService if the CID itself is blocked too).
+
+##### `/ipns/IPNS`
+
+IPNS-rule: Blocks the given IPNS name before resolving. It does not block the CID that it
+resolves to.
+
+If the IPNS name is a domain name, it is blocked directy.
+
+If the IPNS name is a CIDv1 (libp2p-key) or b58-encoded-multihash (CIDV0),
+then the blocking affects the underlying Multihash.
+
+Blocking layer recommendation: NameSystem.
+
+##### `/ipns/IPNS/PATH`
+
+IPNS-Path-rule: Blocks specifically the IPNS path, before resolving. Equivalent to `/ipfs/CID/PATH`.
+
+Blocking layer recommendation: There is no good place to implement this rule
+as the NameSystem only handles IPNS names (without paths), and the
+path.Resolver only handles already-resolved Paths.
+
+##### `/ipns/NAME/*` `/ipns/NAME/PATH*`
+
+IPNS-Path-Prefix-Rule: Same as with the IPFS-Path-Prefix-Rule.
+
+Blocking layer recommendation:  There is no good place to implement this rule
+as the NameSystem only handles IPNS names (without paths), and the
+path.Resolver only handles already-resolved Paths.
+
+##### `/PATH` `/PATH/*` `/PATH*`
+
+Subpath-Rule: Block solely by looking at the subpath component of an IPFS path. Examples:
+
+- `/my/path`: blocks any item that tries to resolve `/my/path`, regardless of the CID used.
+- `/my/path*` and `/my/path/*`: blocks any paths that contain the prefix `/my/path`.
+
+Blocking layer recommendation: PathResolver.
+
+##### `//DOUBLE_HASH`
+
+Doublehash-Rule: Blocks using double-hashed item, which can be:
+
+- The sha256-hex-encoded hash of `CIDV1_BASE32/PATH`: this is the legacy
+  badbits block anchor format. It can only block by CID and not by
+  multihash. When no path present, the trailing slash must be kept
+  (`CIDV1_BASE32/`).
+- A b58-encoded multihash (a.k.a CIDV0), corresponding to the Sum() of:
+  - An IPNS-Path:
+    - `/ipns/IPNS` when the IPNS name is NOT a CID.
+    - The b58-encoded-multihash extracted from an IPNS key when the IPNS key
+      is a CID.
```

Comment:
```suggestion
    - The b58-encoded-multihash extracted from an IPNS name when the IPNS name
      is a CIDv1 with `libp2p-key` codec.
```

Reply:
I don't understand the CIDv1 addition. IPNS names can be CIDv0, in which case there is nothing to do, and CIDv1s? Or should I add CIDv0 as a different line?

I didn't care to research if you can have IPNS names using non b58 encoding for CIDv0 or non libp2p-key for CIDv1...

Reply:
In practice, you try to parse NAME as a CID and if that flies you get the b58-multihash out of it and double hash that.

If not, you double hash `/ipns/name`.

Reply:
This is just a precaution  – better to make spec very explicit and avoid the problems around the fact that PeerID != CIDv0 (CIDv0 only starts with `Q` while PeerID can also start with `1`, see https://cid.ipfs.tech/#12D3KooWHj48PbuAPUf9LeovBJZEXn2A4rseW5z9wbTBwvM82nAq and https://github.com/libp2p/specs/blob/master/peer-ids/peer-ids.md#decoding)